setup.py

```diff
@@ -0,0 +1,64 @@
+from setuptools import setup, find_packages, Extension
+import os, shutil, re
+from pathlib import Path
+from datetime import datetime
+
+# Get the current date and time as version
+def get_version():

```

Comment:
if in the workflow we add the VERSION env variable here we can do something like this:

```
if os.environ.get('VERSION', 'snapshot') == 'snapshot':
    return 'snapshot-' + datetime.now().strftime("%Y%m%d%H%M%S")
else:
    return os.environ['VERSION']
```

Reply:
The standard naming convention for a Python package is package_name-version-distribution.whl. For example, synap-3.1.0-cp310-cp310-linux_x86_64.whl.

Ideally, when we run "pip install synap", it should automatically fetch the latest version of the package from the PyPI repository. However, in our case, we are hosting our wheel on GitHub releases, so we will need to specify the complete URL for the wheel.

If the package name is set to "snapshot", it will appear as "snapshot" in the list of installed packages when using pip, rather than as "synap". During the un-installation process, it needs to be referred to by its package name, so you would uninstall it using "pip uninstall snapshot". It is preferable to maintain the package name as "synap" for clarity and consistency.

Also, the datetime.now().strftime("%Y%m%d%H%M%S") approach is not suitable for pip naming conventions because it generates timestamps that do not provide meaningful versioning information required by package managers like pip.

Reply:
as discussed over the phone I agree what I porposed above is not good.

tensorflow uses the schema X.Y.ZrcW (see https://pypi.org/project/tensorflow/#history), we could do something similar like:

3.1.0snapshotYYMMDD where the first version part comes from the hardcoded version number in the source code as you were doing before.

Reply:
That make sense to use "X.YrcN  # Release Candidate" format as per official standards: https://peps.python.org/pep-0440/

It would be look like this;
synap-3.1.0rc20240703222235-cp310-cp310-linux_x86_64.whl